src/common/components/multiSelectDropdown/MultiSelectDropdown.tsx

```diff
@@ -176,7 +190,11 @@ const Dropdown: React.FC<Props> = ({
 
   const handleValueChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     const val = event.target.value;
-    toggleOption(val);
+    if (val === SELECT_ALL) {
+      handleClear();
```

Comment:
Whats does select all do? seems strange that it clears the array

Reply:
Logic of the "select all" checkbox is described in https://helsinkisolutionoffice.atlassian.net/browse/TH-618 (I agree, it's little bit strange)

Maybe naming is little bit confusing here.